internal/service/gophermart/order/order.go

```diff
@@ -19,30 +18,40 @@ const (
 	StatusProcessed  = "PROCESSED"
 )
 
-type checkJob struct {
-	userID string
-	number string
+type OrderService struct {
+	rep                    gophermart.OrderRepository
+	repTx                  gophermart.WithdrawalRepository
+	accrual                service.AccrualService
+	jobs                   chan checkJob
+	retryOnError           time.Duration
+	attemptsGettingAccrual int
 }
 
-type OrderService struct {
-	rep                  gophermart.OrderRepository
-	repTx                gophermart.WithdrawalRepository
-	accrual              service.AccrualService
-	jobs                 chan checkJob
-	checkerPeriodOnError []time.Duration
+type OrderServiceConfig struct {
+	gophermart.OrderRepository
+	service.AccrualService
+	gophermart.WithdrawalRepository
+	RetryOnError           time.Duration
+	AttemptsGettingAccrual int
 }
 
-func NewOrderService(rep gophermart.OrderRepository, accrual service.AccrualService, repTx gophermart.WithdrawalRepository) OrderService {
+func NewOrderService(ctx context.Context, config OrderServiceConfig) OrderService {
 	s := OrderService{
-		rep:                  rep,
-		repTx:                repTx,
-		accrual:              accrual,
-		jobs:                 make(chan checkJob),
-		checkerPeriodOnError: []time.Duration{10 * time.Second, 1 * time.Minute, 5 * time.Minute},
+		rep:                    config.OrderRepository,
+		repTx:                  config.WithdrawalRepository,
+		accrual:                config.AccrualService,
+		jobs:                   make(chan checkJob),
+		retryOnError:           config.RetryOnError,
+		attemptsGettingAccrual: config.AttemptsGettingAccrual,
 	}
 
 	go s.runCheckerDirector()
```

Comment:
```suggestion
	go s.runCheckerDirector(ctx)
```
У нас в этой функции есть контекст приложения. Почему внутри нее создается свой из воздуха?